### PR TITLE
[FLINK-36616] fix npe in GcpPublisherConfig

### PR DESCRIPTION
**Pull request by stankiewicz**
hey there, 
with current implementation PubSubSinkV2 throws NPE due to transportChannelProvider being null. transportChannelProvider is mostly used for testing.

### CODE REVIEW

**Comment by stankiewicz** [comment]
@vahmed-hamdy would you be able to take a look at this and merge it? 
Also, do you happen to know what is the release cycle for this connector?

**Review by vahmed-hamdy — CHANGES_REQUESTED** [review]
@stankiewicz Thanks a lot for picking this up. 
1- Could you add unit test for this case in `PubSubSinkV2BuilderTest` simply instantiate the configs with null TransportChannel wrapper?
2 - Could you change the commit message to `fix NPE for transportChannelProvider in GcpPublisherConfig` or anything more descriptive?

**Comment by stankiewicz** [comment]
@vahmed-hamdy  done, thanks for the suggestions.

**Review by vahmed-hamdy — APPROVED** [review]
LGTM, thanks for the contribution.
@snuyanzin could you take look here as well please?

File: flink-connector-gcp-pubsub/src/main/java/org/apache/flink/connector/gcp/pubsub/sink/config/GcpPublisherConfig.java [hunk 1]
```diff
@@ -52,7 +52,11 @@ public CredentialsProvider getCredentialsProvider() {
     }
 
     public TransportChannelProvider getTransportChannelProvider() {
-        return transportChannelProvider.getTransportChannelProvider();
+        if (transportChannelProvider==null){
+            return null;
+        } else{
```
Comment (snuyanzin) [inline flink-connector-gcp-pubsub/src/main/java/org/apache/flink/connector/gcp/pubsub/sink/config/GcpPublisherConfig.java]:
```suggestion
        if (transportChannelProvider == null) {
            return null;
        } else {
```
is it able to compile locally? (`./mvnw clean verify`)
I guess it should fail with spotless complains

**Comment by stankiewicz** [comment]
fixed style errors.

File: flink-connector-gcp-pubsub/src/main/java/org/apache/flink/connector/gcp/pubsub/sink/config/GcpPublisherConfig.java [hunk 2]
```diff
@@ -52,7 +52,11 @@ public CredentialsProvider getCredentialsProvider() {
     }
 
     public TransportChannelProvider getTransportChannelProvider() {
+        if (transportChannelProvider == null) {
+            return null;
+        }
         return transportChannelProvider.getTransportChannelProvider();
+
```
Comment (snuyanzin) [inline flink-connector-gcp-pubsub/src/main/java/org/apache/flink/connector/gcp/pubsub/sink/config/GcpPublisherConfig.java]:
```suggestion
```
nit: i guess we don't need this line

**Comment by snuyanzin** [comment]
seems spotless not applied

**Comment by stankiewicz** [comment]
yeah, I don't know why current pom was skipping spotless:apply, I had to forcefully unskip it.

**Review by snuyanzin — APPROVED** [review]

**Comment by snuyanzin** [comment]
Thanks for the contribution @stankiewicz 
thanks for the review @vahmed-hamdy

**Comment by stankiewicz** [comment]
Thanks 👍. Do you know when the next release is planned? Can I help with anything?
